connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntityEquipmentTranslator.java

```diff
@@ -66,17 +66,12 @@ public void translate(GeyserSession session, ServerEntityEquipmentPacket packet)
             ItemData item = ItemTranslator.translateToBedrock(session, equipment.getItem());
             switch (equipment.getSlot()) {
                 case HELMET -> {
-                    CompoundTag profile = null;
-
                     if (livingEntity instanceof PlayerEntity
                             && session.getItemMappings().getMapping(item).getJavaIdentifier().equals("minecraft:player_head")
                             && equipment.getItem().getNbt() != null
                             && equipment.getItem().getNbt().contains("SkullOwner")
                             && equipment.getItem().getNbt().get("SkullOwner") instanceof CompoundTag) {
```

Comment:
This can be simplified even further to match Camotoy's answer. This line can exactly read:
```java
&& equipment.getItem().getNbt().get("SkullOwner") instanceof CompoundTag profile) {
```

Being able to specify a variable on the same line as the `instanceof` is a new feature, you can read more about it here: https://www.baeldung.com/java-pattern-matching-instanceof